std/functional.d

```diff
@@ -306,62 +381,17 @@ private uint _ctfeSkipName(ref string op, string name)
     return 0;
 }
 
-// returns 1 if $(D fun) is trivial unary function
-private uint _ctfeMatchUnary(string fun, string name)
-{
-    if (!__ctfe) assert(false);
-    import std.stdio;
-    fun._ctfeSkipOp();
-    for (;;)
-    {
-        immutable h = fun._ctfeSkipName(name) + fun._ctfeSkipInteger();
-        if (h == 0)
-        {
-            fun._ctfeSkipOp();
-            break;
-        }
-        else if (h == 1)
-        {
-            if(!fun._ctfeSkipOp())
-                break;
-        }
-        else
-            return 0;
-    }
-    return fun.length == 0;
-}
-
-unittest
+// returns 1 if $(D fun) is trivial n-ary function
+private uint _ctfeMatchNary(string fun, string[] names...)
 {
-    static assert(!_ctfeMatchUnary("sqrt(ё)", "ё"));
-    static assert(!_ctfeMatchUnary("ё.sqrt", "ё"));
-    static assert(!_ctfeMatchUnary(".ё+ё", "ё"));
-    static assert(!_ctfeMatchUnary("_ё+ё", "ё"));
-    static assert(!_ctfeMatchUnary("ёё", "ё"));
-    static assert(_ctfeMatchUnary("a+a", "a"));
-    static assert(_ctfeMatchUnary("a + 10", "a"));
-    static assert(_ctfeMatchUnary("4 == a", "a"));
-    static assert(_ctfeMatchUnary("2==a", "a"));
-    static assert(_ctfeMatchUnary("1 != a", "a"));
-    static assert(_ctfeMatchUnary("a!=4", "a"));
-    static assert(_ctfeMatchUnary("a< 1", "a"));
-    static assert(_ctfeMatchUnary("434 < a", "a"));
-    static assert(_ctfeMatchUnary("132 > a", "a"));
-    static assert(_ctfeMatchUnary("123 >a", "a"));
-    static assert(_ctfeMatchUnary("a>82", "a"));
-    static assert(_ctfeMatchUnary("ё>82", "ё"));
-    static assert(_ctfeMatchUnary("ё[ё(ё)]", "ё"));
-    static assert(_ctfeMatchUnary("ё[21]", "ё"));
-}
+    import std.algorithm : reduce;
 
-// returns 1 if $(D fun) is trivial binary function
-private uint _ctfeMatchBinary(string fun, string name1, string name2)
-{
     if (!__ctfe) assert(false);
     fun._ctfeSkipOp();
     for (;;)
     {
-        immutable h = fun._ctfeSkipName(name1) + fun._ctfeSkipName(name2) + fun._ctfeSkipInteger();
+        immutable h = reduce!((a,b) => a + fun._ctfeSkipName(b))(0, names) + fun._ctfeSkipInteger;
```

Comment:
- Please use normal loop instead of reduce. This function must not have dependencies.